lib/index.js

```diff
@@ -143,11 +145,13 @@ class MqttProtocol {
     this.context.debug('[mqtt] onDisconnection %s', client.id);
 
     if (this.connections.has(client)) {
-      const connectionId = this.connections.get(client);
+      setTimeout(() => {
+        const connectionId = this.connections.get(client);
 
-      this.connections.delete(client);
-      delete this.connectionsById[connectionId];
-      this.entryPoint.removeConnection(connectionId);
+        this.connections.delete(client);
+        delete this.connectionsById[connectionId];
+        this.entryPoint.removeConnection(connectionId);
+      }, this.config.disconnectDelay);
     }
```

Comment:
The delay added here is there because when using a CLI client (tried with mosquitto and node mqtt ones), the disconnection is received before the request, which then leads to a situation where the connection id is lost and the request is never sent to Kuzzle.